### PR TITLE
ICMSLST-1943 Add Invite Organisation Contact feature.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Importer and Exporter contacts can now invite other orgainsation contacts.
An email is sent to the new contact who can then decide to accept or reject
the organisation invite.

Changes:
  - Add InviteOrgContactView view.
  - Add AcceptOrgContactInviteView view.
  - Add OrganisationContactInviteEmail GOVNotify email.
  - Update UserFullyRegisteredMiddleware to allow accept invite endpoint.
  - Add tests for new views and email.
  - Fix Exporter Search to exclude agents.
  - Fix contact details in org contact list.

### CODE REVIEW

File: web/templates/partial/contacts/contacts.html [hunk 1]
```diff
@@ -48,11 +69,17 @@ <h3>Contacts</h3>
...
-                <span class="text-widget ">{{ user.email or 'email' }}
-                  <br/> {{ user.organisation or 'organisation' }}
-                  <br/> {{ user.department or 'department' }}
-                  <br/> {{ user.job_title or 'job_title' }}
-                  <br> {{ user.phone_number or 'phone_number' }}
+                <span class="text-widget ">
+                  {{ user.email or "" }}
+                  {% for field in [user.organisation, user.department, user.job_title] %}
+                    {% if field %}
+                      <br/> {{ field }}
```
Comment (mikeylockwood) [inline web/templates/partial/contacts/contacts.html:76]: This html is a bit strange, creating line breaks within a span. Would normally use`<p>` elements in a `<div>`

```
<div class="text-widget">
  {% if user.email %}
    <p>{{ user.email }}</p>
  {% endif %}
  {% for field in [user.organisation, user.department, user.job_title] %}
    {% if field %}
      <p>{{ field }}</p>
     {% endif %}
  {% endfor %}
</div>
```

**Review by mikeylockwood — APPROVED** [review]

File: web/domains/contacts/views.py [hunk 2]
```diff
@@ -77,8 +91,169 @@ def delete(request: AuthenticatedHttpRequest, *, org_type: str, org_pk: int, con
...
+    # FormView config
+    form_class = AcceptOrgInviteForm
+    template_name = "web/domains/organisation/accept-org-invite.html"
+
+    def test_func(self):
+        """Check the invite exists for the logged-in user."""
+
+        # Only match invites linked to the user.
+        importer_invites = ImporterContactInvite.objects.filter(
+            email=self.request.user.email, code=self.kwargs["code"], processed=False
```
Comment (marcuspp) [inline web/domains/contacts/views.py:194]: Unsure why this is a filter as code should be unique and return max 1 row?

File: web/domains/contacts/views.py [hunk 3]
```diff
@@ -77,8 +91,169 @@ def delete(request: AuthenticatedHttpRequest, *, org_type: str, org_pk: int, con
...
+        )
+
+        if importer_invites.exists():
+            return True
+
+        exporter_invites = ExporterContactInvite.objects.filter(
+            email=self.request.user.email, code=self.kwargs["code"], processed=False
+        )
+
+        if exporter_invites.exists():
```
Comment (marcuspp) [inline web/domains/contacts/views.py:204]: `return exporter_invites.exists()`

File: web/domains/contacts/views.py [hunk 4]
```diff
@@ -77,8 +91,169 @@ def delete(request: AuthenticatedHttpRequest, *, org_type: str, org_pk: int, con
...
+        org_pk = self.kwargs["org_pk"]
+
+        model_class = _get_class_imp_or_exp(org_type)
+        org: Org = get_object_or_404(model_class, pk=org_pk)
+
+        return org
+
+    def _get_invite(self) -> type[ExporterContactInvite | ImporterContactInvite]:
+        org_type = self.kwargs["org_type"]
+        if org_type == "importer":
```
Comment (marcuspp) [inline web/domains/contacts/views.py:176]: Consider making "importer" and "exporter" constants

**Review by marcuspp — APPROVED** [review]

File: web/templates/partial/contacts/contacts.html (hunk 1 above)
Comment (MattHolmes123) [reply web/templates/partial/contacts/contacts.html:76]: I just refactored what was already there as I noticed a bug, I will change it and see what it looks like.
It's nothing to do with this pull request really.

**Pull request merged**